src/concrete/ml/quantization/post_training.py

```diff
@@ -23,6 +23,97 @@
 from .quantized_ops import QuantizedBrevitasQuant
 from .quantizers import QuantizationOptions, QuantizedArray, UniformQuantizer
 
+# pylint: disable=too-many-lines
+
+
+def _inspect_tree_n_bits(n_bits):
+    """Validate the 'n_bits' parameter for tree-based models.
+
+    This function checks whether 'n_bits' is a valid integer or dictionary.
+    - If 'n_bits' is an integer, it must be a non-null positive, its value is assigned to
+        "op_inputs" and "op_leaves" bits
+    - If it is a dictionary, it should contain integer values for keys 'op_leaves' and 'op_inputs',
+        where 'op_leaves' should not exceed 'op_inputs'.
+
+    The function raises a ValueError with a descriptive message if 'n_bits' does not meet
+    these criteria.
+
+    Args:
+        n_bits (int, Dict[str, int]): number of bits for quantization, can be a single value or
+            a dictionary with the following keys :
+            - "op_inputs" (mandatory): number of bits to quantize the input values
+            - "op_leaves" (optional): number of bits to quantize the leaves, must be less than or
+                equal to `op_inputs`. defaults to the value of "op_inputs" if not specified.
+
+    Raises:
+        ValueError: If 'n_bits' does not conform to the required format or value constraints.
+    """
+
+    detailed_message = (
+        "Invalid `n_bits`, either pass a non-null positive integer or a dictionary containing "
+        "integer values for the following keys:\n"
+        "- `op_inputs` (mandatory): number of bits to quantize the input values\n"
+        "- `op_leaves` (optional): number of bits to quantize the leaves, must be less than or "
+        "equal to `op_inputs`. Defaults to the value of `op_inputs` if not specified."
+        "When using a single integer for n_bits, its value is assigned to `op_inputs` and "
+        "`op_leaves` bits.\n"
+    )
+
+    error_message = ""
+
+    if isinstance(n_bits, int):
+        if n_bits <= 0:
+            error_message = "n_bits must be a non-null, positive integer"
+    elif isinstance(n_bits, dict):
+        if "op_inputs" not in n_bits.keys() or set(n_bits.keys()) - {"op_leaves", "op_inputs"}:
```

Comment:
small detail but I would maybe separate this into two branches, with `if "op_inputs" not in n_bits.keys()` having an error message like "key 'op_inputs' is mandatory" , which is more precise

Reply:
ok

Reply:
done